superset/reports/commands/execute.py

```diff
@@ -0,0 +1,229 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from datetime import datetime, timedelta
+from typing import Optional
+
+from sqlalchemy.orm import Session
+
+from superset import app, thumbnail_cache
+from superset.commands.base import BaseCommand
+from superset.commands.exceptions import CommandException
+from superset.extensions import security_manager
+from superset.models.reports import (
+    ReportExecutionLog,
+    ReportLogState,
+    ReportSchedule,
+    ReportScheduleType,
+)
+from superset.reports.commands.alert import AlertCommand
+from superset.reports.commands.exceptions import (
+    ReportScheduleAlertGracePeriodError,
+    ReportScheduleExecuteUnexpectedError,
+    ReportScheduleNotFoundError,
+    ReportSchedulePreviousWorkingError,
+    ReportScheduleScreenshotFailedError,
+)
+from superset.reports.dao import ReportScheduleDAO
+from superset.reports.notifications import create_notification
+from superset.reports.notifications.base import NotificationContent, ScreenshotData
+from superset.utils.celery import session_scope
+from superset.utils.screenshots import (
+    BaseScreenshot,
+    ChartScreenshot,
+    DashboardScreenshot,
+)
+from superset.utils.urls import get_url_path
+
+logger = logging.getLogger(__name__)
+
+
+class AsyncExecuteReportScheduleCommand(BaseCommand):
```

Comment:
I read this class a few times looking for suggestions on simplification or breaking it up. I didn't come up with much, but I figured I'd surface that I felt the need to make the effort.

Reply:
Did a slight refactor, main `run` is pretty simple now